src/mission/fly_mission.jl

```diff
@@ -304,4 +280,56 @@ function fly_mission!(ac, imission = 1; itermax = 35, initializes_engine = true)
     end
 
 return 
+end
+
+"""
+    calculate_cruise_altutide_or_CL!(specifying_cruise, WMTO, parg, para, pare, wing, ΔTatmos, imission)
+
+Runs the aircraft through the specified mission, computing and converging the fuel weight. Formerly, `fly_offdesign_mission!()`.
+
+!!! details "🔃 Inputs and Outputs"
+    **Inputs:**
+    - `specifying_cruise::String`: option for whether cruise altitude or lift coefficient is specified. Options are "altitude" or "lift_coefficient"
+    - `WMTO::Float64`: Maximum takeoff weight (N)
+    - `parg::Vector{Float64}`: vector with aircraft geometric parameters
+    - `para::Matrix{Float64}`: array with aerodynamic parameters
+    - `pare::Matrix{Float64}`: array with engine parameters
+    - `wing::Wing`: Wing model object
+    - `ΔTatmos::Float64`: Temperature difference between sea level and standard sea-level (K)
+    - `imission::Int64`: Mission index
+**Outputs:**
+    - No explicit outputs. Computed quantities are saved to `par` arrays of `aircraft` model for the mission selected
+"""
+function calculate_cruise_altutide_or_CL!(specifying_cruise, WMTO, ac, imission)
+    parg, parm, para, pare, _, fuse, _, wing, _, _, _ = unpack_ac(ac, imission)
+
+    #Calculate ΔT for the atmosphere
+    altTO = parm[imaltTO] 
+    T_std,_,_,_,_ = atmos(altTO/1e3)
+    ΔTatmos = parm[imT0TO] - T_std #temperature difference such that T(altTO) = T0TO
+
+    ρ0 = pare[ierho0, ipcruise1]
+    ρcab = max(parg[igpcabin], pare[iep0, ipcruise1]) / (RSL * TSL)
+    WbuoyCR = (ρcab - ρ0) * gee * parg[igcabVol]
+    
+    ip = ipcruise1
+    We = WMTO * para[iafracW, ip]
+    u0 = pare[ieu0, ip]
+    BW = We + WbuoyCR # Weight including buoyancy
+    S = wing.layout.S
+
+    if compare_strings(specifying_cruise, "altitude")   
```

Comment:
A good preemptive error catch might be:

```suggestion
    if compare_strings(specifying_cruise, "altitude")   
        if para[iaCL, ipcruise,1] != para[iaalt, ipcruise, imission] #if CLs are prescribed but 
            @warn "An off-design CL is specified, but the analysis is set to rewrite it. Please review your inputs, especially `specifying_cruise`, if this is not the desired behavior"
        end
```

along with a similar one for the following `elseif`